src/error.rs

```diff
@@ -49,7 +49,72 @@ pub struct Pos {
     path: String,
 }
 
+
+/// This type represents the location that an error occured
+#[derive(Debug)]
+pub struct ErrorMarker {
+    index: usize,
+    line: usize,
+    col: usize,
+} 
+
+impl ErrorMarker {
+    /// The byte index of the error
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    /// The line of the error
+    pub fn line(&self) -> usize {
+        self.line
+    }
+
+    /// The column of the error
+    pub fn col(&self) -> usize {
```

Comment:
I would prefer not to abbreviate "column" because the standard library does not abbreviate it [here](https://doc.rust-lang.org/std/macro.column.html) and [here](https://doc.rust-lang.org/std/panic/struct.Location.html#method.column).